IDE content changes for IA launch (#2187)
dbt-labs/docs.getdbt.com: website/docs/docs/develop/dbt-cloud-features.md

id: "dbt-cloud-features"
---

> ## dbt Cloud IDE

Reviewer: @runleonarun : just a heads up since you own git issue #2074 (dbt Cloud features page). mirna has populated it with IDE-specific content

> :::info Beta Feature

Reviewer: ```suggestion
:::info Join our Cloud IDE beta
```

> The revamped dbt Cloud IDE is now in beta! [Submit your expression of interest](https://docs.google.com/forms/d/e/1FAIpQLSdlU65gqTZPyGAUc16SkxqTc50NO9vdq_KGx1Mjm_4FB_97FA/viewform) to join the new Cloud IDE beta group and review [Develop in the IDE beta](/docs/develop/develop-in-the-cloud#develop-in-the-cloud-ide-beta) to learn more.

Reviewer: ```suggestion
Learn about [what you can do](/docs/develop/develop-in-the-cloud#develop-in-the-cloud-ide-beta) with our Cloud IDE beta. Please [submit your expression of interest](https://docs.google.com/forms/d/e/1FAIpQLSdlU65gqTZPyGAUc16SkxqTc50NO9vdq_KGx1Mjm_4FB_97FA/viewform) to join our Cloud IDE beta group.
```

:::

The dbt Cloud [integrated development environment (IDE)](/docs/develop/develop-in-the-cloud) allows you to build, test, run, and version control your dbt projects directly from your browser. The IDE is the fastest and most reliable way to deploy dbt, and provides a real-time editing and execution environment for your dbt project -- no command line required.  Anyone can use the IDE, from new dbt developers to seasoned practitioners.

With the Cloud IDE, you can:

- Write modular SQL models with `select` statements and the [`ref()`](/docs.getdbt.com/reference/dbt-jinja-functions/ref) function.
- Compile dbt code into SQL and execute it against your database directly.
- Test every model before deploying them to production.
- Generate and view documentation of your dbt project.
- Leverage [git and version-control your code](/docs/collaborate/git/version-control-basics) from your browser with a couple of clicks.

> - Visualize a directed acyclic graph (DAG), [and more](/docs/develop/dbt-cloud-tips)

Reviewer: ```suggestion
- Visualize a directed acyclic graph (DAG) [and more](/docs/develop/dbt-cloud-tips)
```

- Use dark mode when you develop in the Cloud IDE

## IDE features

> The dbt Cloud IDE comes with great features to make it easy for you to develop, build, compile, run and test data models!  Check out some of the features below to learn more:

Reviewer: ```suggestion
The dbt Cloud IDE includes features that make it easier for you to develop, build, compile, run, and test your data models.  Check out some of the features below to learn more:
```

  file or folder name and indicate the actions performed:
       *  Unsaved **(•)** - The IDE detects unsaved changes to your file/folder
       *  Modification **(M)** - The IDE detects a modification of existing files/folders have saved changes file or folder
       *  Untracked **(U)** - The IDE detects changes made to new files or renamed files

- Build, test and run your project with a click of a button, or using the Cloud IDE command bar.

- Drag and drop
  * You can also drag and drop files located in the file explorer. Use the file breadcrumb on the top of the IDE for quick, linear
  navigation. You can access adjacent files in the same file by right clicking on the breadcrumb file.
  
- Organize tabs
  * You can move your tabs around to reorganize your work in the IDE. You can also right click on a tab to view and select a list of actions
  to take.

- Multiple selections
  * You can make multiple selections for quick and simultaneous edits. The below commands are a common way to add more cursors and allow you to insert
  cursors below or above with ease.

     * Option-Command-Down arrow
     * Option-Command-Up arrow

>      * Press Option and click on an area

Reviewer: does it make sense to add a "Related docs" section with a link to the dbt Cloud tips page? i think that might be a topic of interest after reading this page 🤷‍♀️
